src/components/include/utils/threads/message_loop_thread.h

```diff
@@ -148,8 +148,9 @@ MessageLoopThread<Q>::MessageLoopThread(const std::string& name,
 template <class Q>
 MessageLoopThread<Q>::~MessageLoopThread() {
   Shutdown();
-  delete thread_delegate_;
```

Comment:
Thread delegate must be deleted before the thread itself. All threads in the project follow this pattern